src/client/caching/structures/messages/MessageMentions.ts

```diff
@@ -0,0 +1,63 @@
+import { Cache } from '@klasa/cache';
+
+import type { Message } from '../Message';
+import type { APIMessageMentionChannelData, APIChannelData, APIMessageMentionData } from '@klasa/dapi-types';
+import type { User } from '../User';
+import type { Guild } from '../guilds/Guild';
+
+export class MessageMentions {
+
+	/**
+	 * The {@link Message} this entry belongs to.
+	 * @since 0.0.1
+	 */
+	public message: Message;
+
+	/**
+	 * Users specifically mentioned in the message.
+	 * @since 0.0.1
+	 */
+	public users: Cache<string, User> = new Cache();
+
+	/**
+	 * Roles specifically mentioned in this message.
+	 * @since 0.0.1
+	 */
+	public roles: Cache<string, string> = new Cache();
+
+	/**
+	 * Channels specifically mentioned in this message.
+	 * @since 0.0.1
+	 */
+	public channels: Cache<string, APIChannelData> = new Cache();
+
+	/**
+	 * Whether this message mentions everyone.
+	 * @since 0.0.1
+	 */
+	public everyone: boolean;
+
+	public constructor(message: Message, users: APIMessageMentionData[], roles: string[], channels: APIMessageMentionChannelData[], everyone: boolean) {
+		this.message = message;
+
+		if (users) {
```

Comment:
Preeety sure this won't work since the caches are constructed before the if check